submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,338 @@
+let FRIENDS_ARRAY = [],
+	INITIAL_FRIENDS_ARRAY = [];
+const USERS_AMOUNT = 24,
+	API_URL = `https://randomuser.me/api/?results=${USERS_AMOUNT}`,
+	CARDS_CONTAINER = document.querySelector(".cards__container"),
+	TOTAL_COUNTER = document.querySelector(".amount");
+
+function getFriends() {
+	fetch(API_URL)
+		.then((response) => {
+			if (checkResponseStatus(response.status)) {
+				return response.json();
+			} else {
+				appendErrorMessage(
+					getResponseErrorMessage(
+						response.status,
+						response.statusText
+					)
+				);
+			}
+		})
+		.then((responseBody) => {
+			if (responseBody !== undefined) {
+				INITIAL_FRIENDS_ARRAY = INITIAL_FRIENDS_ARRAY.concat(
+					flattenFriendProperties(responseBody.results)
+				);
+				FRIENDS_ARRAY = INITIAL_FRIENDS_ARRAY;
+				appendFriendsCards(FRIENDS_ARRAY);
+				setTotalCounter(FRIENDS_ARRAY);
+				initializeAgeLimits(FRIENDS_ARRAY);
+			}
+		})
+		.catch((error) =>
+			appendErrorMessage(`${error} <br> Try to reload the page!`)
+		);
+}
+
+getFriends();
+
+function checkResponseStatus(status) {
+	if (status >= 200 && status < 300) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+function setTotalCounter(friendsArray) {
+	TOTAL_COUNTER.innerText = `${friendsArray.length} Totals`;
+	if (!friendsArray.length) {
+		appendNoResultsMessage();
+	}
+}
+
+function initializeAgeLimits(friendsArray) {
+	["minAge", "maxAge"].forEach((limitInputId) =>
+		["min", "max"].forEach((limitType) =>
+			setAgeLimit(
+				getCertainAgeLimit(friendsArray, limitType),
+				limitInputId,
+				limitType
+			)
+		)
```

Comment:
These arrays are redundant. You should avoid such stuff in the future.

Please, replace it with code without such arrays.

Reply:
Thank you for your advice! Actually, I didn't like the way it looked:
```
setAgeLimit(getCertainAgeLimit(friendsArray, 'min'), 'minAge', 'min');
setAgeLimit(getCertainAgeLimit(friendsArray, 'max'), 'minAge', 'max');
setAgeLimit(getCertainAgeLimit(friendsArray, 'min'), 'maxAge', 'min');
setAgeLimit(getCertainAgeLimit(friendsArray, 'max'), 'maxAge', 'max');
```
Therefore, I've decided to make the setting of elements attributes kinda autonomous. But now I see that this was not the best idea ever gotten :sweat_smile:
I've replaced the code without such arrays.